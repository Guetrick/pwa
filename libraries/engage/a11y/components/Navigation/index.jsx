import React from 'react';
import PropTypes from 'prop-types';
import { Link, I18n } from '@shopgate/engage/components';
import styles from './style';

/**
 * @param {Object} props component props.
 * @param {string} props.title Navigation title.
 * @param {Array} props.entries Navigation entries.
 * @param {Object} context  The component context.
 * @returns {JSX}
 */
const Navigation = ({ title, entries }, context) => {
  const { __ } = context.i18n();
  if (!entries || !entries.length) {
    return null;
  }

  return (
    <nav aria-label={title ? __(title) : null}>
      <ul className={styles}>
        {entries.map(({ title: entryTitle, link }) => (
          <li key={link}>
            <Link href={link}>
              <I18n.Text string={entryTitle} />
            </Link>
          </li>
        ))}
      </ul>
    </nav>
  );
};

Navigation.propTypes = {
  entries: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string,
    link: PropTypes.string,
  })),
  title: PropTypes.string,
};

Navigation.defaultProps = {
  title: null,
  entries: [],
};

Navigation.contextTypes = {
  i18n: PropTypes.func,
};

export default Navigation;
