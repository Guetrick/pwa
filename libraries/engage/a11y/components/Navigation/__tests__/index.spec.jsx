import React from 'react';
import { mount } from 'enzyme';
import Navigation from '../index';

jest.mock('../../../../components', () => ({
  I18n: {
    Text: function Translate({ string }) {
      return string;
    },
  },
  Link: function Link({ children }) { return children; },
}));

const mockTranslate = jest.fn(key => key);

const mockRenderOptions = {
  context: {
    i18n: () => ({
      __: mockTranslate,
    }),
  },
};

const entries = [{
  title: 'Title One',
  link: '/some-link',
}, {
  title: 'Title Two',
  link: '/another-link',
}];

describe('<Navigation />', () => {
  it('should render with an aria label and entries', () => {
    const title = 'Navigation Title';
    const wrapper = mount((<Navigation title={title} entries={entries} />), mockRenderOptions);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('nav').prop('aria-label')).toEqual(title);
    expect(wrapper.find('ul').children()).toHaveLength(2);
  });

  it('should render empty when no entries are passed', () => {
    const wrapper = mount((<Navigation />), mockRenderOptions);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.isEmptyRender()).toBe(true);
  });
});
