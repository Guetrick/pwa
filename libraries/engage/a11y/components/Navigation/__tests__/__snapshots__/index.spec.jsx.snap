// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should render empty when no entries are passed 1`] = `
<Navigation
  entries={Array []}
  title={null}
/>
`;

exports[`<Navigation /> should render with an aria label and entries 1`] = `
<Navigation
  entries={
    Array [
      Object {
        "link": "/some-link",
        "title": "Title One",
      },
      Object {
        "link": "/another-link",
        "title": "Title Two",
      },
    ]
  }
  title="Navigation Title"
>
  <nav
    aria-label="Navigation Title"
  >
    <ul
      className={
        Object {
          "data-css-1krt3b8": "",
        }
      }
    >
      <li
        key="/some-link"
      >
        <Link
          href="/some-link"
        >
          <Translate
            string="Title One"
          >
            Title One
          </Translate>
        </Link>
      </li>
      <li
        key="/another-link"
      >
        <Link
          href="/another-link"
        >
          <Translate
            string="Title Two"
          >
            Title Two
          </Translate>
        </Link>
      </li>
    </ul>
  </nav>
</Navigation>
`;
